javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -43,8 +43,9 @@ protected static void generateProtocolMethodJavadoc(ClientMethod clientMethod, J
                     // ignore if synthesized by modelerfour, i.e. api-version
                     .filter(p -> p.getOrigin() == ParameterSynthesizedOrigin.NONE)
                     .collect(Collectors.toList());
-            if (!queryParameters.isEmpty()) {
+            if (!queryParameters.isEmpty() && hasParametersToPrintInJavadoc(queryParameters)) {
                 optionalParametersJavadoc("Query Parameters", queryParameters, commentBlock);
+                commentBlock.line("You can add these to a request with {@code new RequestOptions.addQueryParam()}");
```

Comment:
Recommend to use `{@link RequestOptions#addQueryParam}`, as not every parameter require a `new RequestOptions`